tests/endpoints/bookstore_grpc/server/server.go

```diff
@@ -266,6 +267,21 @@ func (s *BookstoreServerV1Impl) CreateShelf(ctx context.Context, req *bspbv1.Cre
 		return nil, err
 	}
 
+	// Unmarshal, deepcopy and marshal, to verify the received binary `any`
+	if req.Shelf.Any != nil {
+		var obj bspbv1.ObjectOnlyForAny
+		err := ptypes.UnmarshalAny(req.Shelf.Any, &obj)
+		if err != nil {
+			return nil, status.New(codes.Internal, "cannot unmarshal shelf.any").Err()
+		}
+
+		newObj := obj
```

Comment:
I don't think this will make a deep copy. It will copy the struct, but it wont copy members that are pointers inside the struct.

I don't think a deep copy is really needed, but would be nice to have. Otherwise, update the comment on line 270.